deflate.h

```diff
@@ -373,20 +368,22 @@ void ZLIB_INTERNAL bi_windup(deflate_state *s);
 
 # define _tr_tally_lit(s, c, flush) \
   { unsigned char cc = (c); \
-    s->d_buf[s->last_lit] = 0; \
-    s->l_buf[s->last_lit++] = cc; \
+    s->sym_buf[s->sym_next++] = 0; \
+    s->sym_buf[s->sym_next++] = 0; \
```

Comment:
This is bad...

Reply:
What is bad?
I see you posted a bunch of those comments, but no info about what is bad about it and why it is bad, and what could be done to improve on it.

Reply:
@Dead2 These lines cause unnecessary slowdown compared to the previous version of the code.

Reply:
So there is nothing wrong with this patch, except that it could be followed up with an optimization patch.

Reply:
@Dead2 Wrong.